src/server/transaction.cc

```diff
@@ -1194,6 +1200,32 @@ bool Transaction::NotifySuspended(TxId committed_txid, ShardId sid) {
   return false;
 }
 
+void Transaction::LogJournalOnShard(EngineShard* shard) {
+  // TODO: For now, we ignore non shard coordination.
+  if (shard == nullptr)
+    return;
+
+  if ((cid_->opt_mask() & CO::WRITE) == 0)
+    return;
+
+  auto journal = shard->journal();
+  if (journal == nullptr)
+    return;
+
+  // TODO: Handle complex commands like LMPOP correctly once they are implemented.
```

Comment:
Currently, we don't have commands with tail arguments (arguments after we've listed all per-shard values). Lets see how they'll be implemented and according to it we'll do it here.